### ISSUE TITLE
Add copy_dir

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3387]
`copy_files` currently only copy files (even if you use `*` or `**`). 

It would be great if we could copy a full subtree, using something like `copy_dir`.

**Comment by diml** [comment]
Indeed. Note that `**` is not for recursive globs. This has surprised several users, but this is actually documented: https://dune.readthedocs.io/en/stable/concepts.html#glob

We are thinking of adding support for recursive globs, and maybe in `(lang dune 3)` we'll change the meaning of `**`.
